src/Action.cpp

```diff
@@ -13,6 +13,7 @@ Action::Action(const Conditions& preconditions, const Conditions& postconditions
 
 Action::Action(const Name& name, const Conditions& preconditions, const Conditions& postconditions)
     : _name { name },
+      _id { std::hash<std::string>{}(name) },
```

Comment:
I doubt this strongly guarantees key uniqueness

Reply:
I join you in your doubt. I'm probably better off just [implementing it myself](http://www.cse.yorku.ca/~oz/hash.html).